src/util/notifications_util.js

```diff
@@ -127,25 +129,21 @@ class Notificator {
                 let connection = this.connect_str_to_connection.get(notif.meta.name);
                 if (!connection) {
                     connection = get_connection(connect);
-                    try {
-                        await connection.connect();
-                    } catch (err) {
-                        //failed to connect
-                        dbg.error("Connection failed for", connect);
-                        await failure_append(str);
-                        return;
-                    }
+                    await connection.connect();
                     this.connect_str_to_connection.set(notif.meta.name, connection);
                 }
-                const send_promise = connection.promise_notify(notif, failure_append);
+                const send_promise = connection.promise_notify(notif, this.handle_failed_notification.bind(this), failure_append);
                 if (send_promise) send_promises.push(send_promise);
                 if (send_promises.length > this.batch_size) {
                     await Promise.all(send_promises);
                     send_promises = [];
                 }
             } catch (err) {
                 dbg.error("Failed to notify. err = ", err, ", str =", str);
-                await failure_append(str);
+                //re-write the failed notification if it's still configured on the bucket
+                if (notif) {
```

Comment:
No need of this check. 
"notif" will not be null at this point. If it could be, then we should have checked it before line no 124.

Reply:
notif can be null in case json parsing fails (currently line 123).
If json parsing is ok, notif will not be null on 124 and forward.